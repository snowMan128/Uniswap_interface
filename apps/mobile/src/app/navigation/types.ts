import {
  CompositeNavigationProp,
  CompositeScreenProps,
  NavigatorScreenParams,
  useNavigation,
} from '@react-navigation/native'
import { NativeStackNavigationProp, NativeStackScreenProps } from '@react-navigation/native-stack'
import { EducationContentType } from 'src/components/education'
import { HomeScreenTabIndex } from 'src/screens/HomeScreenTabIndex'
import { FiatOnRampScreens, OnboardingScreens, Screens, UnitagScreens } from 'src/screens/Screens'
import { NFTItem } from 'wallet/src/features/nfts/types'
import { ImportType, OnboardingEntryPoint } from 'wallet/src/features/onboarding/types'

type NFTItemScreenParams = {
  owner?: Address
  address: string
  tokenId: string
  isSpam?: boolean
  fallbackData?: NFTItem
}

export type CloudBackupFormParams = {
  address: Address
  password: string
}

export type ExploreStackParamList = {
  [Screens.Explore]: undefined
  [Screens.ExternalProfile]: {
    address: string
  }
  [Screens.NFTItem]: NFTItemScreenParams
  [Screens.NFTCollection]: { collectionAddress: string }
  [Screens.TokenDetails]: {
    currencyId: string
  }
}

export type FiatOnRampStackParamList = {
  [FiatOnRampScreens.AmountInput]: undefined
  [FiatOnRampScreens.ServiceProviders]: undefined
}

export type SettingsStackParamList = {
  [Screens.Dev]: undefined
  [Screens.Settings]: undefined
  [Screens.SettingsAppearance]: undefined
  [Screens.SettingsBiometricAuth]: undefined
  [Screens.SettingsCloudBackupPasswordConfirm]: CloudBackupFormParams
  [Screens.SettingsCloudBackupPasswordCreate]: { address: Address }
  [Screens.SettingsCloudBackupProcessing]: CloudBackupFormParams
  [Screens.SettingsCloudBackupStatus]: { address: Address }
  [Screens.SettingsHelpCenter]: undefined
  [Screens.SettingsLanguage]: undefined
  [Screens.SettingsPrivacy]: undefined
  [Screens.SettingsViewSeedPhrase]: { address: Address; walletNeedsRestore?: boolean }
  [Screens.SettingsWallet]: { address: Address }
  [Screens.SettingsWalletEdit]: { address: Address }
  [Screens.SettingsWalletManageConnection]: { address: Address }
  [Screens.WebView]: { headerTitle: string; uriLink: string }
}

export type OnboardingStackBaseParams = {
  importType: ImportType
  entryPoint: OnboardingEntryPoint
}

export type OnboardingStackParamList = {
  [OnboardingScreens.BackupManual]: OnboardingStackBaseParams
  [OnboardingScreens.BackupCloudPasswordCreate]: {
    address: Address
  } & OnboardingStackBaseParams
  [OnboardingScreens.BackupCloudPasswordConfirm]: CloudBackupFormParams & OnboardingStackBaseParams
  [OnboardingScreens.BackupCloudProcessing]: CloudBackupFormParams & OnboardingStackBaseParams
  [OnboardingScreens.Backup]: OnboardingStackBaseParams
  [OnboardingScreens.Landing]: OnboardingStackBaseParams
  [OnboardingScreens.EditName]: OnboardingStackBaseParams
  [OnboardingScreens.Notifications]: OnboardingStackBaseParams
  [OnboardingScreens.QRAnimation]: OnboardingStackBaseParams
  [OnboardingScreens.Security]: OnboardingStackBaseParams

  // unitag
  [UnitagScreens.ClaimUnitag]: { entryPoint: OnboardingScreens.Landing | Screens.Home }
  [UnitagScreens.ChooseProfilePicture]: {
    entryPoint: OnboardingScreens.Landing | Screens.Home
    unitag: string
  }

  // import
  [OnboardingScreens.ImportMethod]: OnboardingStackBaseParams
  [OnboardingScreens.RestoreCloudBackupLoading]: OnboardingStackBaseParams
  [OnboardingScreens.RestoreCloudBackup]: OnboardingStackBaseParams
  [OnboardingScreens.RestoreCloudBackupPassword]: {
    mnemonicId: string
  } & OnboardingStackBaseParams
  [OnboardingScreens.SeedPhraseInput]: OnboardingStackBaseParams
  [OnboardingScreens.SelectWallet]: OnboardingStackBaseParams
  [OnboardingScreens.WatchWallet]: OnboardingStackBaseParams
}

export type UnitagStackParamList = {
  [UnitagScreens.ClaimUnitag]: { entryPoint: OnboardingScreens.Landing | Screens.Home }
  [UnitagScreens.ChooseProfilePicture]: {
    entryPoint: OnboardingScreens.Landing | Screens.Home
    unitag: string
  }
  [UnitagScreens.UnitagConfirmation]: {
    unitag: string
    address: Address
    profilePictureUri?: string
  }
  [UnitagScreens.EditProfile]: {
    address: Address
    unitag: string
    entryPoint: UnitagScreens.UnitagConfirmation | Screens.SettingsWallet
  }
}

export type AppStackParamList = {
  [Screens.Education]: {
    type: EducationContentType
  } & OnboardingStackBaseParams
  [Screens.Home]?: { tab?: HomeScreenTabIndex }
  [Screens.OnboardingStack]: NavigatorScreenParams<OnboardingStackParamList>
  [Screens.SettingsStack]: NavigatorScreenParams<SettingsStackParamList>
  [Screens.UnitagStack]: NavigatorScreenParams<UnitagStackParamList>
  [Screens.TokenDetails]: {
    currencyId: string
  }
  [Screens.NFTItem]: NFTItemScreenParams
  [Screens.NFTCollection]: { collectionAddress: string }
  [Screens.ExternalProfile]: {
    address: string
  }
  [Screens.WebView]: { headerTitle: string; uriLink: string }
}

export type AppStackNavigationProp = NativeStackNavigationProp<AppStackParamList>
export type AppStackScreenProps = NativeStackScreenProps<AppStackParamList>
export type AppStackScreenProp<Screen extends keyof AppStackParamList> = NativeStackScreenProps<
  AppStackParamList,
  Screen
>

export type ExploreStackNavigationProp = CompositeNavigationProp<
  NativeStackNavigationProp<ExploreStackParamList>,
  AppStackNavigationProp
>

export type SettingsStackNavigationProp = CompositeNavigationProp<
  NativeStackNavigationProp<SettingsStackParamList>,
  AppStackNavigationProp
>

export type SettingsStackScreenProp<Screen extends keyof SettingsStackParamList> =
  CompositeScreenProps<NativeStackScreenProps<SettingsStackParamList, Screen>, AppStackScreenProps>

export type OnboardingStackNavigationProp = CompositeNavigationProp<
  NativeStackNavigationProp<OnboardingStackParamList>,
  AppStackNavigationProp
>

export type UnitagStackScreenProp<Screen extends keyof UnitagStackParamList> =
  NativeStackScreenProps<UnitagStackParamList, Screen>

export type RootParamList = AppStackParamList &
  ExploreStackParamList &
  OnboardingStackParamList &
  SettingsStackParamList &
  UnitagStackParamList &
  FiatOnRampStackParamList

export const useAppStackNavigation = (): AppStackNavigationProp =>
  useNavigation<AppStackNavigationProp>()
export const useExploreStackNavigation = (): ExploreStackNavigationProp =>
  useNavigation<ExploreStackNavigationProp>()
export const useSettingsStackNavigation = (): SettingsStackNavigationProp =>
  useNavigation<SettingsStackNavigationProp>()
export const useOnboardingStackNavigation = (): OnboardingStackNavigationProp =>
  useNavigation<OnboardingStackNavigationProp>()
