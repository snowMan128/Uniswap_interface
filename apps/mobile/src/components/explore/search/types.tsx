import { SearchResult } from 'wallet/src/features/search/SearchResult'
import { SEARCH_RESULT_HEADER_KEY } from './constants'

// Header type used to render header text instead of SearchResult item

export type SearchResultOrHeader =
  | SearchResult
  | { type: typeof SEARCH_RESULT_HEADER_KEY; title: string }
