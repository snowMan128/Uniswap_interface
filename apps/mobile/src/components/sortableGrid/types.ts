import { FlatList, ScrollView, View } from 'react-native'
import { SharedValue } from 'react-native-reanimated'

export type Require<T, K extends keyof T = keyof T> = Required<Pick<T, K>> & Omit<T, K>

export type ItemMeasurements = {
  height: number
  width: number
  x: number
  y: number
}

export type AutoScrollProps = {
  scrollableRef: React.RefObject<FlatList | ScrollView>
  visibleHeight: SharedValue<number>
  scrollY: SharedValue<number>
  // The parent container inside the scrollable that wraps the grid
  // (e.g. when the grid is rendered inside the FlatList header)
  // if not provided, we assume that the grid is the first child in
  // the scrollable container
  containerRef?: React.RefObject<View>
}

export type SortableGridContextType = {
  gridContainerRef: React.RefObject<View>
  itemAtIndexMeasurements: SharedValue<ItemMeasurements[]>
  dragActivationProgress: SharedValue<number>
  activeIndex: number | null
  previousActiveIndex: SharedValue<number | null>
  activeTranslation: SharedValue<{ x: number; y: number }>
  scrollOffsetDiff: SharedValue<number>
  renderIndexToDisplayIndex: SharedValue<number[]>
  setActiveIndex: (index: number | null) => void
  onDragStart?: () => void
  displayToRenderIndex: SharedValue<number[]>
  activeItemScale: SharedValue<number>
  visibleHeight: SharedValue<number>
  activeItemOpacity: SharedValue<number>
  activeItemShadowOpacity: SharedValue<number>
  touchedIndex: SharedValue<number | null>
  editable: boolean
  containerStartOffset: SharedValue<number>
  containerEndOffset: SharedValue<number>
}

export type SortableGridRenderItemInfo<I> = {
  item: I
  index: number
  dragActivationProgress: SharedValue<number>
  isTouched: SharedValue<boolean>
}

export type SortableGridRenderItem<I> = (info: SortableGridRenderItemInfo<I>) => JSX.Element

export type Vector = {
  x: number
  y: number
}

export type SortableGridChangeEvent<I> = {
  data: I[]
  fromIndex: number
  toIndex: number
}
