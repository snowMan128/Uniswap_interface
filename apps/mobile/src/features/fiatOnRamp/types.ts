import { CurrencyInfo } from 'wallet/src/features/dataApi/types'

export type FiatOnRampCurrency = {
  currencyInfo: Maybe<CurrencyInfo>
  moonpayCurrencyCode?: string
}

export enum InitialQuoteSelection {
  MostRecent,
  Best,
}
