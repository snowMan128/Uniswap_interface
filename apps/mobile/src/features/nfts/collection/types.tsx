import { NftCollectionScreenQuery } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'

export type NFTCollectionData = Maybe<
  NonNullable<NonNullable<NftCollectionScreenQuery['nftCollections']>['edges']>[0]['node']
>
