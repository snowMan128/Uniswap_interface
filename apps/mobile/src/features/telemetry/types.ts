import { RenderPassReport } from '@shopify/react-native-performance'
import { SharedEventName } from '@uniswap/analytics-events'
import { MobileEventName, ShareableEntity } from 'src/features/telemetry/constants'
import { WidgetEvent, WidgetType } from 'src/features/widgets/widgets'
import { TraceProps } from 'utilities/src/telemetry/trace/Trace'
import { ImportType } from 'wallet/src/features/onboarding/types'
import { CurrencyField } from 'wallet/src/features/transactions/transactionState/types'
import { EthMethod, WCEventType, WCRequestOutcome } from 'wallet/src/features/walletConnect/types'

// Events related to Moonpay internal transactions
// NOTE: we do not currently have access to the full life cycle of these txs
// because we do not yet use Moonpay's webhook
export type MoonpayTransactionEventProperties = TraceProps &
  // allow any object of strings for now
  Record<string, string>

export type AssetDetailsBaseProperties = {
  name?: string
  address: string
  chain?: number
}

export type SearchResultContextProperties = {
  category?: string
  query?: string
  suggestion_count?: number
  position?: number
  isHistory?: boolean
}

type OnboardingCompletedProps = {
  wallet_type: ImportType
  accounts_imported_count: number
  wallets_imported: string[]
  cloud_backup_used: boolean
}

export type MobileEventProperties = {
  [MobileEventName.AppRating]: {
    type: 'store-review' | 'feedback-form' | 'remind'
    appRatingPromptedMs?: number
    appRatingProvidedMs?: number
  }
  [MobileEventName.BalancesReport]: {
    total_balances_usd: number
    wallets: string[]
    balances: number[]
  }
  [MobileEventName.DeepLinkOpened]: {
    url: string
    screen: 'swap' | 'transaction'
    is_cold_start: boolean
  }
  [MobileEventName.ExploreFilterSelected]: {
    filter_type: string
  }
  [MobileEventName.ExploreSearchResultClicked]: SearchResultContextProperties &
    AssetDetailsBaseProperties & {
      type: 'collection' | 'token' | 'address'
    }
  [MobileEventName.ExploreTokenItemSelected]: AssetDetailsBaseProperties & {
    position: number
  }
  [MobileEventName.FavoriteItem]: AssetDetailsBaseProperties & {
    type: 'token' | 'wallet'
  }
  [MobileEventName.FiatOnRampQuickActionButtonPressed]: TraceProps
  [MobileEventName.FiatOnRampBannerPressed]: TraceProps
  [MobileEventName.FiatOnRampAmountEntered]: TraceProps & { source: 'chip' | 'textInput' }
  [MobileEventName.FiatOnRampWidgetOpened]: TraceProps & { externalTransactionId: string }
  [MobileEventName.OnboardingCompleted]: OnboardingCompletedProps & TraceProps
  [MobileEventName.PerformanceReport]: RenderPassReport
  [MobileEventName.PerformanceGraphql]: {
    dataSize: number
    duration: number
    operationName: string
    operationType?: string
  }
  [MobileEventName.ShareButtonClicked]: {
    entity: ShareableEntity
    url: string
  }
  [MobileEventName.ShareLinkOpened]: {
    entity: ShareableEntity
    url: string
  }
  [MobileEventName.TokenDetailsOtherChainButtonPressed]: TraceProps
  [MobileEventName.TokenSelected]: TraceProps &
    AssetDetailsBaseProperties &
    SearchResultContextProperties & {
      field: CurrencyField
    }
  [MobileEventName.WalletAdded]: OnboardingCompletedProps & TraceProps
  [MobileEventName.WalletConnectSheetCompleted]: {
    request_type: WCEventType
    eth_method?: EthMethod
    dapp_url: string
    dapp_name: string
    wc_version: string
    connection_chain_ids?: number[]
    chain_id?: number
    outcome: WCRequestOutcome
  }
  [MobileEventName.WidgetConfigurationUpdated]: WidgetEvent
  [MobileEventName.WidgetClicked]: {
    widget_type: WidgetType
    url: string
  }
  [SharedEventName.APP_LOADED]: TraceProps | undefined
  [SharedEventName.ELEMENT_CLICKED]: TraceProps
  [SharedEventName.PAGE_VIEWED]: TraceProps
}
