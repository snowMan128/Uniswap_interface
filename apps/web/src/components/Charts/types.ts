import { SeriesDataItemTypeMap, SeriesOptionsMap, Time } from 'lightweight-charts'

export type SeriesDataItemType = SeriesDataItemTypeMap<Time>[keyof SeriesOptionsMap]
