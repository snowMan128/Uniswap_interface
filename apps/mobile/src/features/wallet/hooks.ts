import { useCallback, useEffect, useState } from 'react'
import { useAppSelector } from 'src/app/hooks'
import { openModal } from 'src/features/modals/modalSlice'
import { selectModalState } from 'src/features/modals/selectModalState'
import { FeatureFlags } from 'uniswap/src/features/gating/flags'
import { useFeatureFlag } from 'uniswap/src/features/gating/hooks'
import { logger } from 'utilities/src/logger/logger'
import { Keyring } from 'wallet/src/features/wallet/Keyring/Keyring'
import { useNativeAccountExists } from 'wallet/src/features/wallet/hooks'
import { useAppDispatch } from 'wallet/src/state'
import { ModalName } from 'wallet/src/telemetry/constants'

export function useWalletRestore(params?: { openModalImmediately?: boolean }): {
  walletNeedsRestore: undefined | boolean
  openWalletRestoreModal: () => void
  isModalOpen: boolean
} {
  const dispatch = useAppDispatch()
  const openModalImmediately = params?.openModalImmediately
  // Means that no private key found for mnemonic wallets
  const [walletNeedsRestore, setWalletNeedsRestore] = useState<boolean>(false)
  const hasImportedSeedPhrase = useNativeAccountExists()
  const isRestoreWalletEnabled = useFeatureFlag(FeatureFlags.RestoreWallet)

  const openWalletRestoreModal = useCallback((): void => {
    dispatch(openModal({ name: ModalName.RestoreWallet }))
  }, [dispatch])

  useEffect(() => {
    if (!hasImportedSeedPhrase || !isRestoreWalletEnabled) {
      return
    }

    const openRestoreWalletModalIfNeeded = async (): Promise<void> => {
      const addresses = await Keyring.getAddressesForStoredPrivateKeys()
      setWalletNeedsRestore(hasImportedSeedPhrase && !addresses.length)
    }
    openRestoreWalletModalIfNeeded().catch((error) =>
      logger.error(error, { tags: { file: 'wallet/hooks', function: 'useWalletRestore' } })
    )
  }, [dispatch, hasImportedSeedPhrase, isRestoreWalletEnabled])

  useEffect(() => {
    if (openModalImmediately && walletNeedsRestore) {
      openWalletRestoreModal()
    }
  }, [openModalImmediately, openWalletRestoreModal, walletNeedsRestore])

  const isModalOpen = useAppSelector(selectModalState(ModalName.RestoreWallet)).isOpen

  return { walletNeedsRestore, openWalletRestoreModal, isModalOpen }
}
