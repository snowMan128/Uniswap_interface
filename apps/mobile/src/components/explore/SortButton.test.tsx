import ContextMenu from 'react-native-context-menu-view'
import { render } from 'src/test/test-utils'
import { TokenSortableField } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'
import { ClientTokensOrderBy } from 'wallet/src/features/wallet/types'
import { SortButton } from './SortButton'

jest.mock('react-native-context-menu-view', () => {
  // Use the actual implementation of `react-native-context-menu-view` as the mock implementation
  // (we use mock just to get the props of the component in test)
  return jest.fn(jest.requireActual('react-native-context-menu-view').default)
})

describe('SortButton', () => {
  it('renders without error', () => {
    const tree = render(<SortButton orderBy={TokenSortableField.Volume} />)

    expect(tree).toMatchSnapshot()
  })

  const cases = [
    { test: 'volume', orderBy: TokenSortableField.Volume, label: 'Volume' },
    { test: 'total value locked', orderBy: TokenSortableField.TotalValueLocked, label: 'TVL' },
    { test: 'market cap', orderBy: TokenSortableField.MarketCap, label: 'Market cap' },
    {
      test: 'price increase',
      orderBy: ClientTokensOrderBy.PriceChangePercentage24hDesc,
      label: 'Price increase',
    },
    {
      test: 'price decrease',
      orderBy: ClientTokensOrderBy.PriceChangePercentage24hAsc,
      label: 'Price decrease',
    },
  ]

  describe.each(cases)('when ordering by $test', ({ orderBy, label }) => {
    it(`renders ${label} as the selected option`, () => {
      const { queryByText } = render(<SortButton orderBy={orderBy} />)
      const selectedOption = queryByText(label)

      expect(selectedOption).toBeTruthy()
    })

    it(`returns correct context menu actions with checmark near the ${label} option`, () => {
      jest.clearAllMocks()
      render(<SortButton orderBy={orderBy} />)

      expect((ContextMenu as unknown as jest.Mock).mock.calls[0][0]).toEqual(
        expect.objectContaining({
          actions: [
            {
              title: 'Uniswap volume (24H)',
              systemIcon: orderBy === TokenSortableField.Volume ? 'checkmark' : '',
              orderBy: TokenSortableField.Volume,
            },
            {
              title: 'Uniswap TVL',
              systemIcon: orderBy === TokenSortableField.TotalValueLocked ? 'checkmark' : '',
              orderBy: TokenSortableField.TotalValueLocked,
            },
            {
              title: 'Market cap',
              systemIcon: orderBy === TokenSortableField.MarketCap ? 'checkmark' : '',
              orderBy: TokenSortableField.MarketCap,
            },
            {
              title: 'Price increase (24H)',
              systemIcon:
                orderBy === ClientTokensOrderBy.PriceChangePercentage24hDesc ? 'checkmark' : '',
              orderBy: ClientTokensOrderBy.PriceChangePercentage24hDesc,
            },
            {
              title: 'Price decrease (24H)',
              systemIcon:
                orderBy === ClientTokensOrderBy.PriceChangePercentage24hAsc ? 'checkmark' : '',
              orderBy: ClientTokensOrderBy.PriceChangePercentage24hAsc,
            },
          ],
        })
      )
    })
  })
})
