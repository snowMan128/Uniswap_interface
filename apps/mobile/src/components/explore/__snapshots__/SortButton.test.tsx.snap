// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton renders without error 1`] = `
<ContextMenu
  actions={
    [
      {
        "orderBy": "VOLUME",
        "systemIcon": "checkmark",
        "title": "Uniswap volume (24H)",
      },
      {
        "orderBy": "TOTAL_VALUE_LOCKED",
        "systemIcon": "",
        "title": "Uniswap TVL",
      },
      {
        "orderBy": "MARKET_CAP",
        "systemIcon": "",
        "title": "Market cap",
      },
      {
        "orderBy": "PriceChangePercentage24hDesc",
        "systemIcon": "",
        "title": "Price increase (24H)",
      },
      {
        "orderBy": "PriceChangePercentage24hAsc",
        "systemIcon": "",
        "title": "Price decrease (24H)",
      },
    ]
  }
  dropdownMenuMode={true}
  onPress={[Function]}
>
  <View
    cancelable={true}
    disabled={false}
    focusable={true}
    hitSlop={[Function]}
    minPressDuration={0}
    onBlur={[Function]}
    onFocus={[Function]}
    onLongPress={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 999999,
        "borderBottomRightRadius": 999999,
        "borderTopLeftRadius": 999999,
        "borderTopRightRadius": 999999,
        "flexDirection": "row",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 4,
        }
      }
    >
      <Text
        allowFontScaling={true}
        lineBreakMode="clip"
        maxFontSizeMultiplier={1.2}
        numberOfLines={1}
        style={
          {
            "color": "#7D7D7D",
            "flexShrink": 1,
            "fontFamily": "Basel-Medium",
            "fontSize": 17,
            "fontWeight": "500",
            "lineHeight": 24,
          }
        }
        suppressHighlighting={true}
      >
        Volume
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomLeftRadius": 999999,
            "borderBottomRightRadius": 999999,
            "borderTopLeftRadius": 999999,
            "borderTopRightRadius": 999999,
            "flexDirection": "column",
            "justifyContent": "center",
            "transform": [
              {
                "rotate": "270deg",
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="20"
          bbWidth="20"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeWidth={8}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#7D7D7D",
                "height": 20,
                "width": 20,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor="#7D7D7D"
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeWidth="8"
          >
            <RNSVGPath
              d="M15 6L9 12L15 18"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="3"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</ContextMenu>
`;
